packages/app/src/cli/services/app-context.ts

```diff
@@ -127,3 +142,26 @@ async function logMetadata(app: {apiKey: string}, organization: Organization, re
     cmd_app_reset_used: resetUsed,
   }))
 }
+
+/**
+ * This function loads an app locally without making any network calls.
+ * It uses local specifications and doesn't require the app to be linked.
+ *
+ * @returns The local app instance.
+ */
+export async function localAppContext({
+  directory,
+  userProvidedConfigName,
+  unsafeReportMode = false,
+}: LocalAppContextOptions): Promise<AppInterface> {
+  // Load local specifications only
+  const specifications = await loadLocalExtensionsSpecifications()
+
+  // Load the local app using the specifications
+  return loadApp({
+    directory,
+    userProvidedConfigName,
+    specifications,
+    mode: unsafeReportMode ? 'report' : 'strict',
+  })
+}
```

Comment:
I think we should explicitly name this as something "unsafe", because `localAppContext` is very similar to `loadAppContext` and can be confusing.

I'd rename it to `unsafeLocalAppContext` or `unsafeLoadApp` (since it only returns the app)

Reply:
I don't like using `unsafe` here, it's perfectly safe to use it... But I agree it can be confusing to know what to return. I think [this suggestion](https://github.com/Shopify/cli/pull/5979#discussion_r2149303459) would help.